rest-service/manager_rest/test/endpoints/test_deployment_labels_dependencies.py

```diff
@@ -1,613 +1,420 @@
+        self._deployment(id='env3')
+        self._deployment(id='env4')
+        self.client.deployments.update_labels(
+            'env1',
+            labels=[{'csys-obj-type': 'environment'}]
+        )
+        for env_name in ['env2', 'env3']:
+            self.client.deployments.update_labels(
+                env_name,
+                labels=[
+                    {'csys-obj-type': 'environment'},
+                    {'csys-obj-parent': 'env1'}
+                ]
+            )
+        self.client.deployments.update_labels(
+            'env4',
+            labels=[
+                {'csys-obj-type': 'environment'},
+                {'csys-obj-parent': 'env2'}
+            ]
         )
-
-        self._create_deployment_objects('env_1', 'environment', 2)
-        deployment = self.client.deployments.get('env')
-        self.assertEqual(deployment.sub_environments_count, 5)
-        deployment = self.client.deployments.get('env',
-                                                 all_sub_deployments=False)
+        deployment = self.client.deployments.get('env1')
         self.assertEqual(deployment.sub_environments_count, 3)
-
-    def test_add_sub_deployments_after_deployment_update(self):
-        _, _, _, deployment = self.put_deployment(
-            deployment_id='env',
-            blueprint_id='env'
-        )
-        _, _, _, deployment_1 = self.put_deployment(
-            deployment_id='env_1',
-            blueprint_id='env_1'
-        )
-
-        self.assertEqual(deployment.sub_services_count, 0)
-        self.assertEqual(deployment.sub_services_count, 0)
-        self.assertEqual(deployment_1.sub_environments_count, 0)
-        self.assertEqual(deployment_1.sub_environments_count, 0)
-
-        self.put_deployment(deployment_id='sub_srv', blueprint_id='srv')
-        self.put_deployment_with_labels(
-            [
-                {
-                    'csys-obj-type': 'environment',
-                }
-            ],
-            resource_id='sub_env'
-        )
-
-        self.put_blueprint(
-            blueprint_id='update_sub_srv',
-            blueprint_file_name='blueprint_with_parent_labels.yaml'
-        )
-        self.put_blueprint(
-            blueprint_id='update_sub_env',
-            blueprint_file_name='blueprint_with_parent_labels.yaml'
-        )
-
-        self.client.deployment_updates.update_with_existing_blueprint(
-            'sub_srv', blueprint_id='update_sub_srv'
-        )
-        self.client.deployment_updates.update_with_existing_blueprint(
-            'sub_env', blueprint_id='update_sub_env'
-        )
-        deployment = self.client.deployments.get('env')
-        deployment_1 = self.client.deployments.get('env_1')
-        self.assertEqual(deployment.sub_services_count, 1)
-        self.assertEqual(deployment.sub_services_count, 1)
-        self.assertEqual(deployment_1.sub_environments_count, 1)
-        self.assertEqual(deployment_1.sub_environments_count, 1)
```

Comment:
deployment-update tests can f.. right off. Especially ones that use .yaml blueprints, on this level. If you want to test that, that goes in inte-tests.

Deployment update (the new one) works in terms of updating labels via the restservice, anyway, which is covered by all the other tests already.